use-cases/virgo/hpo.py

```diff
@@ -0,0 +1,225 @@
+import argparse
+import os
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import ray
+import torch
+from ray import train, tune
+
+from itwinai.parser import ConfigParser
+
+
+def run_trial(config):
+    """
+    Execute a single trial using the given configuration (config).
+    This runs a full training pipeline - you can also specify a pipeline as a dictionary, 
+    e.g. if you only want to run certain parts without changing your config.yaml file (see below).
+
+    Args:
+    - config: Dictionary with hyperparameters (e.g., 'batch_size', 'lr').
+
+    Example to run with a manual pipeline (you would have to import these classes):
+
+    my_pipeline = Pipeline(
+        [
+            TimeSeriesDatasetSplitter(
+                train_proportion=0.9,
+                root_folder="/p/scratch/intertwin/datasets/virgo"
+            ),
+            NoiseGeneratorTrainer(
+                config=config,
+                num_epochs=4,
+                strategy=None,
+                checkpoint_path='checkpoints/checkpoint_epoch_{}.pth',
+                validation_every=20
+            )
+        ]
+    )
+
+    Passing a seed to TimeSeriesDatasetSplitter and NoiseGeneratorTrainer
+    will make runs uniform across trials
+    (reducing the variablility to the hyperparameter settings)
+    """
+
+    parser = ConfigParser(
+        config=Path('config.yaml'),
+        override_keys={
+            # Set HPOs controlled by ray
+            'batch_size': config['batch_size'],
+            'learning_rate': config['lr'],
+            # Override logger field, because performance is logged by ray
+            'training_pipeline.init_args.steps.2.init_args.logger': None
+        }
+    )
+    my_pipeline = parser.parse_pipeline(
+        pipeline_nested_key='training_pipeline',
+        verbose=False
+    )
+
+    # Skip the first step of the pipeline (data generation)
+    my_pipeline[1:].execute()
+
+
+def run_hpo(args):
+    """
+    Run hyperparameter optimization using Ray Tune.
+    Either starts a new optimization run or resumes from previous results.
+
+    Args:
+    - args: Command-line arguments parsed by argparse.
+    """
+    if not args.load_old_results:
```

Comment:
This whole section has a bunch of comments that might be redundant. The comments from the other hpo file apply here as well (as it is nearly identical)

Reply:
Perhaps this whole function could be reused if it is all the same?